docs/build_pipeline_controller.md

```diff
@@ -11,16 +11,23 @@ flowchart TD
 
   %% Node definitions
 predicate((PREDICATE: <br> Filter events related to <br> PipelineRuns <br> proccessed by Chains <br> that have <br> succeeded))
-get_pipeline_run{Pipeline found?}
+new_pipeline_run{Pipeline created?}
+get_pipeline_run{Pipeline updated?}
```

Comment:
1. I think the controller doesn't respond to PLR updates, instead, it responds to build PLRs that succeeded.

2. Maybe we need one more branch that describes the "Pipeline failed?" scenario.

Reply:
I had also forgotten to add a `remove finalizer` step to the existing diagram.  So I added that step and linked it to both the build PLR succeeded control flow and the build PLR failed control flow.

Reply:
Thanks!
Just curious, is the text `Pipeline updated?` correct, or it should be `Pipeline succeeded?` instead?